### PR TITLE
New Mir Ion pasrser implementation 

### PR DESCRIPTION
**Pull request by 9il**
Please update the table then. It should be faster then Asdf, I hope so.

### CODE REVIEW

File: json/test_mir_ion_file.d [hunk 1]
```diff
@@ -1,18 +0,0 @@
...
-dflags "-mcpu=native" "-linkonce-templates" "-enable-cross-module-inlining" platform="ldc"
-+/
-import mir_common_json;
-import mir.stdio;
-import mir.deser.json : deserializeJsonFile;
-
-void main() @safe @nogc
-{
-    "D/ldc2 (Mir Amazon's Ion DOM, file input)".notifyStart;
-    auto coordinate = "/tmp/1.json".deserializeJsonFile!Avg.coordinates.avg;
```
Comment (9il) [inline json/test_mir_ion_file.d:15]: The function is no longer supported.

File: json/mir-common-json/source/mir_common_json.d [hunk 2]
```diff
@@ -49,6 +49,7 @@ auto notifyStop(scope const char[] msg = "stop")
     notify(msg);
 }
 
+@serdeIgnoreUnexpectedKeys
```
Comment (nuald) [inline json/mir-common-json/source/mir_common_json.d:52]: Could you please clarify why did you add this line (and the same below)?

Comment (9il) [reply json/mir-common-json/source/mir_common_json.d:52]: Unlike the old one, the new implementation will throw exceptions in case of keys that hasn't been handled by the structure. The UDAs are used to preserve the old behaviour and do not throw.

Comment (9il) [reply json/mir-common-json/source/mir_common_json.d:52]: For example, do not throw exception, that this input isn't used:
```
      "name": "wxetbf 1966",
      "opts": {
        "1": [
          1,
          true
        ]
      }
```

**Comment by nuald** [comment]
Unfortunately, the newer version is even slower. Please ensure that you use the same LDC version (1.29.0) - I've already seen the cases when the compiler upgrade regressed the tests performance.

**Comment by 9il** [comment]
Haha, how bad it is?

**Comment by nuald** [comment]
Please see #415 (new 231ms vs old 192ms).
